src/BinSkim.Rules/EnableSpectreMitigations.cs

```diff
@@ -238,50 +245,73 @@ public override void Analyze(BinaryAnalyzerContext context)
                     continue;
                 }
 
-                SwitchState QSpectreState = SwitchState.SwitchNotFound;
-                SwitchState d2guardspecloadState = SwitchState.SwitchNotFound;
+
+                string[] mitigationSwitches = new string[] { "/Qspectre", "/guardspecload" };
+
+                SwitchState effectiveState;
 
                 // Go process the command line to check for switches
-                if (supportsQspectre)
-                {
-                    QSpectreState = omDetails.GetSwitchState("/Qspectre", OrderOfPrecedence.LastWins);
-                }
+                effectiveState = omDetails.GetSwitchStateWithAliases(mitigationSwitches, onbyDefault == true ? SwitchState.SwitchEnabled : SwitchState.SwitchDisabled, OrderOfPrecedence.LastWins);
 
-                if(supportsd2guardspecload)
+                if (effectiveState == SwitchState.SwitchDisabled)
                 {
-                    // /d2xxxx options show up in the PDB without the d2 string
-                    // So search for just /guardspecload
-                    d2guardspecloadState = omDetails.GetSwitchState("/guardspecload", OrderOfPrecedence.LastWins);
-                }
+                    SwitchState QSpectreState = SwitchState.SwitchNotFound;
+                    SwitchState d2guardspecloadState = SwitchState.SwitchNotFound;
 
-                // TODO-paddymcd-MSFT: Check all the /O optimization flags to determine if we are /Od or not
-                // /Od may disable the Spectre Mitigations.
+                    if (supportsQspectre)
+                    {
+                        QSpectreState = omDetails.GetSwitchState(mitigationSwitches[0] /*"/Qspectre"*/ , OrderOfPrecedence.LastWins);
+                    }
 
-                SwitchState effectiveState = SwitchState.SwitchNotFound;
+                    if (supportsd2guardspecload)
+                    {
+                        // /d2xxxx options show up in the PDB without the d2 string
+                        // So search for just /guardspecload
+                        d2guardspecloadState = omDetails.GetSwitchState(mitigationSwitches[1] /*"/guardspecload"*/, OrderOfPrecedence.LastWins);
+                    }
 
-                // if either QSpectre or d2guardspecload are enabled AND neither is explicitly disabled then we are protected
-                //      (use of both is confusing so issue an error in this scenario even though they are effectively the same switch)
-                if ((QSpectreState == SwitchState.SwitchEnabled || d2guardspecloadState == SwitchState.SwitchEnabled) && 
-                    (QSpectreState != SwitchState.SwitchDisabled && d2guardspecloadState != SwitchState.SwitchDisabled))
-                {
-                    effectiveState = SwitchState.SwitchEnabled;
-                }
+                    if (QSpectreState == SwitchState.SwitchNotFound && d2guardspecloadState == SwitchState.SwitchNotFound)
```

Comment:
Nit/Style:  Either have both continues on the inside of the blocks in this if/else, or have one after the if/else is done.

Reply:
good catch - will correct